metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/ObjectsImporter.java

```diff
@@ -118,12 +140,15 @@ public Analysis importAnalysis(String accession) {
             AnalysisType analysisType = sraAnalysisXmlParser.parseXml(xml, accession);
             analysis = analysisConverter.convert(analysisType);
             analysis.setFiles(fileExtractorFromAnalysis.getFiles(analysisType));
+
             List<ReferenceSequence> referenceSequences = new ArrayList<>();
-            for (String referenceSequenceAccession : getReferenceSequenceAccessions(analysisType)) {
-                //TODO ReferenceSequence Import
-                //referenceSequences.add(importReferenceSequence(referenceSequenceAccession));
+            ReferenceSequence referenceSequence = buildReferenceSequence(analysisType);
```

Comment:
ImportReferenceSequence is not called at all  ?

Reply:
In ImportReferenceSequence() => IsraXmlRetrieverByAccession.getXml is selecting analysis record with reference sequence accession and fails. If i am correct on slack we discussed this and I interpreted DB call should not be for ref seq. Please let me know

Reply:
yes, you need to override importReferenceSequence in SraImporterDatabase to manually create refSequence instead calling super.

Reply:
sure

Reply:
Is there a way to get reference sequence name from its accession? (From analysisType ref seq accession is retrieved and getting ref name is inside this is straight forward)

Reply:
@iyangar it should be possible, although accessions are not always present and they are not always RefSeq accessions. Missing refname is one of the edge cases which I outlined in the [spreadsheet](https://docs.google.com/spreadsheets/d/18IT7defkFXC0zkwhO0kQyGWOi1UbMmN9SjJq6B3yHfc/edit?usp=sharing), let's see what decision is made concerning it.

Reply:
This is implemented in SraObjectsImporterThroughDatabase. This comment was made before i implemented by saving both accession and ref name.